File: src/java/org/javaturk/dp/ch05/flyweight/book/Character.java
package org.javaturk.dp.ch05.flyweight.book;

public class Character {
	// Intrinsic properties
	private char value;
	private boolean upperCase;

	// Extrinsic properties
	private Line line;
	private int position;

	public Character(char value, boolean upperCase) {
		this.value = value;
		this.upperCase = upperCase;
	}
	
	public Character(int value) {
		this.value = (char) value;
	}

	public char getValue() {
		return value;
	}
	
	public boolean isUpperCase() {
		return upperCase;
	}

	public Line getLine() {
		return line;
	}

	public void setLine(Line line) {
		this.line = line;
	}

	public int getPosition() {
		return position;
	}

	public void setPosition(int position) {
		this.position = position;
	}
	
	public String toString() {
		return String.valueOf(value);
	}
}

File: src/typescript/com/gokselkucuksahin/dp/structural-patterns/flyweight/book/character.ts
import {Line} from "./line";

export class Character {
  // Intrinsic properties
  private _value: string;
  private _upperCase: boolean;

  // Extrinsic properties
  private _line: Line;
  private _position: number;

  private constructor(value: string) {
    this._value = value;
  }

  static character_1(value: string) {
    return new Character(value);
  }

  static character_2(value: string, upperCase: boolean) {
    const char = new Character(value);
    char._upperCase = upperCase;
    return char;
  }

  get value(): string {
    return this._value;
  }

  set value(value: string) {
    this._value = value;
  }

  get upperCase(): boolean {
    return this._upperCase;
  }

  set upperCase(value: boolean) {
    this._upperCase = value;
  }

  get line(): Line {
    return this._line;
  }

  set line(value: Line) {
    this._line = value;
  }

  get position(): number {
    return this._position;
  }

  set position(value: number) {
    this._position = value;
  }

  toString(): string {
    return this.value.valueOf().normalize().toString();
  }
}